pipenv/utils/resolver.py

```diff
@@ -723,19 +732,21 @@ def _get_hashes_from_pypi(self, ireq):
             return None
 
     def collect_hashes(self, ireq):
-        if ireq.link:
-            link = ireq.link
-            if link.is_vcs or (link.is_file and link.is_existing_dir()):
```

Comment:
Factor out this base case of `set()` and move the other case to the end of this method for reasons that involve applying index restrictions first.  We will also consider the `ireq.link` preferable to the `ireq.original_link` in the resulting refactor since adding in treatment of the `ireq.link` was important for this example edge case.